### DISCUSSION TITLE
Lightning Channel in "Closing" Status for days, can't do anything else.. Please help!

### DISCUSSION THREAD

**Discussion by khmu007** [Ride-The-Lightning/RTL#1341] [General]
Using Umbrel Home, I closed a lightning channel (LC) on the Lightning Node App. The LC status says "Closing", I see all the info and my Sats in the LC, but my Lightning wallet is decreased to 0 Sats and it hasn't showed up in my BTC wallet. I can't do anything else to the LC.  Is there a way to find the "Closing" LC on the RTL app so I can see what's going on? How can I find the transaction ID to look up in the Mempool, because I don't see one? Any advice is appreciated, thank you.

![Screenshot LC Closing Status](https://github.com/Ride-The-Lightning/RTL/assets/156029055/e69d5b49-4bbc-4fd3-9dca-162372829963)

**Comment by saubyk** [comment]
Hi @khmu007 couple of questions:
1. Was it a cooperative close of a force close? Do you know if the channel was inactive/active when you closed it?
2. What do you see when you log into RTL on the `Peers/Channel` page? You can get to it by clicking on `Peers/Channels` menu under `Lightning`. You should look at the `Pending` or the `Closed` tab

**Reply by saubyk** [reply]
Also, did your channel had substantial balance on the local side, which you're expecting to get back onchain?

**Reply by khmu007** [reply]
Hi and thank you for the quick response. See below answers, also I have to use the Default App Password every time I log into RTL, is that normal? and title on top of the screen says Ride The Lightning - VIOLENTBOUNCE-v23.08. Don't know if it's suppose to be that or something else?

1. It was a force close as I did it myself. It was an active channel, but offline (don't know what that means though).
2. I see Open 0, Pending/Inactive 0, Active HTLCs 0 ; Don't see a Closed Tab.
3. Yes. My local balance is 5,005,530 Sats, as per the attached screen shot I sent above.

**Reply by khmu007** [reply]
Hi Saubyk, just following up to see if you had any updates. I did run command  ~/umbrel/scripts/app compose lightning exec lnd lncli pendingchannels  It shows a closing_txid: 85c5fbeca40f0a5b94805299919d0eedf3b50f7247aeb48e4c186dbef357e0ac  Trying to figure out how to bump the fee in command prompt. Any advice? thanks..

**Reply by saubyk** [reply]
hey there, couple of things:
1. If the channel was force closed, the funds take some time to get back to you. Depends on the CSV setting of the channel
2. I couldn't find the above transaction in the mempool. Can you share the funding transaction id of the channel? That way we should be able to trace if the channel's closing transaction made it to the mempool or not
3. Can you post the complete output of the `pendingchannels` command here

**Reply by khmu007** [reply]
See below output. I also ran  ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpfee 85c5fbeca40f0a5b94805299919d0eedf3b50f7247aeb48e4c186dbef357e0ac:0 --sat_per_vbyte 40  to change the fee, but got an ERROR message.

 "total_limbo_balance": "5005530",
    "pending_open_channels": [],
    "pending_closing_channels": [],
    "pending_force_closing_channels": [],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02e0d304ea61db67dcee3c3c4b53b1c43e3ed0a52e369cc86fe6e3562579739aa7",
                "channel_point": "f540a1042bbe18a00d81e58b97e80f045b34f4738b0a06f91a8334b423c36fc2:1",
                "capacity": "5009000",
                "local_balance": "5005530",
                "remote_balance": "0",
                "local_chan_reserve_sat": "50090",
                "remote_chan_reserve_sat": "50090",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusBorked|ChanStatusCommitBroadcasted|ChanStatusLocalCloseInitiator",
                "private": false,
                "memo": ""
            },
            "limbo_balance": "5005530",
            "commitments": {
                "local_txid": "85c5fbeca40f0a5b94805299919d0eedf3b50f7247aeb48e4c186dbef357e0ac",
                "remote_txid": "0a4c9a9e083c3380132dfd1d530a0db34030f66bcc1e02d4a71a724953b9fb72",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2810",
                "remote_commit_fee_sat": "2810",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": "85c5fbeca40f0a5b94805299919d0eedf3b50f7247aeb48e4c186dbef357e0ac"

**Reply by saubyk** [reply]
If you look at your channel funding transaction in the mempool, you can see that the output funding your channel is still unspent:
https://mempool.space/tx/f540a1042bbe18a00d81e58b97e80f045b34f4738b0a06f91a8334b423c36fc2

Which means that either your channel closing transaction failed to broadcast, or was broadcast but eventually dropped from the mempool. You can see that the below transaction (supposedly the channel closing txn) is not in the mempool:
https://mempool.space/tx/85c5fbeca40f0a5b94805299919d0eedf3b50f7247aeb48e4c186dbef357e0ac

Given that status of the channel now is `ChanStatusBorked`, you can try the below steps, in order to attempt to rebroadcast the channel closing transaction:
1. Ensure that your node's mempool size is at least 1GB, by setting the `maxmempool=1000` in the config
2. Restart the node
3. Once you are able to rebroadcast the channel closing transaction, you should be able to bump it up further to hasten the channel close

**Reply by khmu007** [reply]
In Advanced Settings, which Tab do I go in to change it to maxmempool=1000? See pic, I looked in all 6 Tabs and didn't see it, thanks..
![Advance Settings](https://github.com/Ride-The-Lightning/RTL/assets/156029055/2fa980f1-9476-4543-8e07-d2e7f16146ce)

**Reply by saubyk** [reply]
maxmempool is a bitcoin setting not lightning. Look for the below in your bitcoin `advanced settings` under the `optimization` section
![image](https://github.com/Ride-The-Lightning/RTL/assets/39208279/503a04d0-7a50-43e2-9940-d0e0f35148fe)

**Reply by khmu007** [reply]
I changed the mempool size and restarted the node, but still wasn't able to bump the fee. Umbrel has a video on X about an Automatic Channel Recovery to recover funds. Basically, it's uninstalling, then reinstalling the lightning node using your 24 recovery words, and selecting the time your node was last online. The funds could take up to 2 days to show up in your BTC wallet, but it never did. I ran the output of pending channels again and it's very different from before. My "total_limbo_balance" and "local balance" are now 0 and no "closing_txid". Any guidance is appreciated, thanks..

"total_limbo_balance": "0",
    "pending_open_channels": [],
    "pending_closing_channels": [],
    "pending_force_closing_channels": [],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02e0d304ea61db67dcee3c3c4b53b1c43e3ed0a52e369cc86fe6e3562579739aa7",
                "channel_point": "f540a1042bbe18a00d81e58b97e80f045b34f4738b0a06f91a8334b423c36fc2:1",
                "capacity": "5009000",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "0",
                "chan_status_flags": "ChanStatusRestored",
                "private": true,
                "memo": ""
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""

**Reply by saubyk** [reply]
Hi @khmu007 is your channel active again?

**Reply by khmu007** [reply]
No, it's still in "Closing" status.

**Reply by saubyk** [reply]
> Basically, it's uninstalling, then reinstalling the lightning node using your 24 recovery words, and selecting the time your node was last online. The funds could take up to 2 days to show up in your BTC wallet, but it never did

Did you attempt a recovery process for your node? The status flag `ChanStatusRestored` indicates that channel funds are getting restored from the backup

Another thing you can try is reach out (if possible) to your peer and ask them to force close this channel

**Reply by khmu007** [reply]
Yes I did, using the Automatic Channel Recover method, but the Sats still aren't showing up in my BTC wallet and it's been 3 days since I did it. With the status flag saying "ChanStatusRestored", does that means it'll show up in my BTC wallet eventually?

Also, what do I give/tell my peer so they can force close this channel?  The channel point, my node ID, etc.

**Reply by saubyk** [reply]
I wish you didn't attempt to recover the node, as I was hoping to figure out a way to broadcast the channel close transaction. And I did advise you to just restart.

Anyway, If the status has been `ChanStatusRestored` for the last 3 days, then only thing left to do is contact your peer, give them your pubkey and request them to force close the channel/s they have with your node

If none of this works, you can work with chantools to recover your funds: https://github.com/lightninglabs/chantools
But I don't have expertise in that, and you'll have to seek help (in case you need it) separately.

Hope this helps

**Reply by khmu007** [reply]
Ok, will do.. Thanks for trying and helping out, I really appreciate it. Is the pubkey my Node ID?

**Comment by saubyk** [comment]
> Ok, will do.. Thanks for trying and helping out, I really appreciate it. Is the pubkey my Node ID?

yeah pubkey is the node id
